tests/Unit/Comment/CommentContainerTest.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sabberworm\CSS\Tests\Unit\Comment;
+
+use PHPUnit\Framework\Constraint\LogicalAnd;
+use PHPUnit\Framework\Constraint\TraversableContains;
+use PHPUnit\Framework\TestCase;
+use Sabberworm\CSS\Comment\Comment;
+use Sabberworm\CSS\Comment\Commentable;
+use Sabberworm\CSS\Tests\Unit\Comment\Fixtures\ConcreteCommentContainer;
+use TRegx\DataProvider\DataProviders;
+
+/**
+ * @covers \Sabberworm\CSS\Comment\CommentContainer
+ */
+final class CommentContainerTest extends TestCase
+{
+    /**
+     * @var Commentable
+     */
+    private $subject;
+
+    protected function setUp(): void
+    {
+        $this->subject = new ConcreteCommentContainer();
+    }
+
+    /**
+     * @test
+     */
+    public function getCommentsInitiallyReturnsEmptyArray(): void
+    {
+        self::assertSame([], $this->subject->getComments());
+    }
+
+    /**
+     * @return array<non-empty-string, array{0: list<Comment>}>
+     */
+    public function provideCommentArray(): array
+    {
+        return [
+            'no comment' => [[]],
+            'one comment' => [[new Comment('Is this really a spoon?')]],
+            'two comments' => [[
+                new Comment('I’m a teapot.'),
+                new Comment('I’m a cafetière.'),
+            ]],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param list<Comment> $comments
+     *
+     * @dataProvider provideCommentArray
+     */
+    public function getCommentsAfterCommentsAddedToVirginContainerReturnsThoseComments(array $comments): void
+    {
+        $this->subject->addComments($comments);
+
+        self::assertSame($comments, $this->subject->getComments());
+    }
+
+    /**
+     * @test
+     *
+     * @param list<Comment> $comments
+     *
+     * @dataProvider provideCommentArray
+     */
+    public function getCommentsAfterEmptyArrayOfCommentsAddedReturnsOriginalComments(array $comments): void
```

Comment:
Maybe we can describe the effects instead of focusing on the methods:

```suggestion
    public function setCommentsWithEmptyArrayKeepsOriginalCommentsUnchanged(array $comments): void
```

Reply:
Apart from `getCommentsInitiallyReturnsEmptyArray()`, all the test methods focus on either `addComments()` (first group) or `setComments()` (last group).  So perhaps they all me renamed like `addCommentsDoesXYZ()` and `setCommentsDoesXYZ` - WDYT?

Reply:
Yes, I'd like that!

Reply:
Done.